File: pulsar-skeleton/src/main/java/fun/platonic/pulsar/crawl/parse/ParseResult.java
/*******************************************************************************
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ******************************************************************************/
package fun.platonic.pulsar.crawl.parse;

import org.jsoup.nodes.Document;
import fun.platonic.pulsar.persist.HypeLink;
import fun.platonic.pulsar.persist.ParseStatus;

import javax.annotation.Nonnull;
import java.util.ArrayList;

public class ParseResult extends ParseStatus {

    private ArrayList<HypeLink> hypeLinks = new ArrayList<>();
    private Document document;

    public ParseResult() {
        super(ParseStatus.NOTPARSED, ParseStatus.SUCCESS_OK);
    }

    public ParseResult(short majorCode, int minorCode) {
        super(majorCode, minorCode);
    }

    public ParseResult(short majorCode, int minorCode, String message) {
        super(majorCode, minorCode, message);
    }

    @Nonnull
    public static ParseResult failed(int minorCode, String message) {
        return new ParseResult(FAILED, minorCode, message);
    }

    @Nonnull
    public static ParseResult failed(Throwable e) {
        return new ParseResult(FAILED, FAILED_EXCEPTION, e.getMessage());
    }

    public Document getDocument() {
        return document;
    }

    public void setDocument(Document document) {
        this.document = document;
    }

    public ArrayList<HypeLink> getHypeLinks() {
        return hypeLinks;
    }
}

File: pulsar-skeleton/src/main/kotlin/ai/platon/pulsar/crawl/parse/ParseResult.kt
/*******************************************************************************
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package ai.platon.pulsar.crawl.parse

import ai.platon.pulsar.common.FlowState
import ai.platon.pulsar.common.urls.LabeledHyperlink
import ai.platon.pulsar.persist.HyperlinkPersistable
import ai.platon.pulsar.persist.ParseStatus
import ai.platon.pulsar.persist.metadata.ParseStatusCodes
import ai.platon.pulsar.persist.model.DomStatistics
import java.util.concurrent.ConcurrentSkipListSet
import kotlin.reflect.KClass

class ParseResult(
        majorCode: Short = NOTPARSED,
        minorCode: Int = SUCCESS_OK,
        message: String? = null
) : ParseStatus(majorCode, minorCode, message) {
    val hypeLinks = mutableSetOf<HyperlinkPersistable>()
    var domStatistics: DomStatistics? = null
    var parsers = mutableListOf<KClass<out Parser>>()
    var flowStatus = FlowState.CONTINUE

    val shouldContinue get() = flowStatus == FlowState.CONTINUE
    val shouldBreak get() = flowStatus == FlowState.BREAK

    companion object {
        val labeledHypeLinks = ConcurrentSkipListSet<LabeledHyperlink>()

        fun failed(minorCode: Int, message: String?): ParseResult {
            return ParseResult(ParseStatusCodes.FAILED, minorCode, message)
        }

        fun failed(e: Throwable): ParseResult {
            return ParseResult(ParseStatusCodes.FAILED, ParseStatusCodes.FAILED_EXCEPTION, e.message)
        }
    }
}
